[GTK][WPE] Deprecate WebKitConsoleMessage API

carlosgcampos opened the pull request:
#### d671caa0409902d3ac1cc879073c83c5131d41dc
<pre>
[GTK][WPE] Deprecate WebKitConsoleMessage API
<a href="https://bugs.webkit.org/show_bug.cgi?id=247791">https://bugs.webkit.org/show_bug.cgi?id=247791</a>

Reviewed by Michael Catanzaro.

The API seems to be unused so we can also remove the implementation.

* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitConsoleMessage.cpp:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitConsoleMessage.h.in:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebPage.cpp:
(webkit_web_page_class_init):
(webkitWebPageCreate):
(webkitWebPageDidSendConsoleMessage): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestConsoleMessage.cpp: Removed.
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitWebContext.cpp:
(xhrMessageReceivedCallback):
(testWebContextSecurityFileXHR):
(consoleMessageReceivedCallback): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitGLib/WebExtensionTest.cpp:
(pageCreatedCallback):
(consoleMessageSentCallback): Deleted.
* Tools/TestWebKitAPI/glib/CMakeLists.txt:

Canonical link: <a href="https://commits.webkit.org/257010@main">https://commits.webkit.org/257010@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/43fc0353a0cc5dc84be4965b31abc5061281ffea

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/97521 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/6776 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/30687 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/107025 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/167289 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/101478 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/7114 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/35532 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/89919 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/103694 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/103176 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/5328 "Found 1 new test failure: fast/forms/ios/focus-button.html (failure)") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/84153 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/32334 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/87201 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/88993 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/75259 "Found 4 new API test failures: /WebKitGTK/TestConsoleMessage:/webkit/WebKitConsoleMessage/console-api, /WebKitGTK/TestConsoleMessage:/webkit/WebKitConsoleMessage/security-error, /WebKitGTK/TestConsoleMessage:/webkit/WebKitConsoleMessage/js-exception, /WebKitGTK/TestConsoleMessage:/webkit/WebKitConsoleMessage/network-error (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/778 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/20457 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/766 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/21933 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/5579 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/44404 "Found 30 new test failures: animations/stop-animation-on-suspend.html, fast/images/animated-heics-draw.html, http/tests/navigation/parsed-iframe-dynamic-form-back-entry.html, http/tests/privateClickMeasurement/send-attribution-conversion-request.html, http/wpt/service-workers/cross-site-navigation-same-cookie-lax.https.html, http/wpt/service-workers/fetch-service-worker-preload-download.https.html, imported/w3c/web-platform-tests/css/css-flexbox/flexbox_flex-1-0-Npercent.html, imported/w3c/web-platform-tests/css/css-masking/clip-path-svg-content/clip-path-no-content-003.svg, imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-rotated-will-change-transform.html, imported/w3c/web-platform-tests/css/css-masking/mask-image/mask-image-3c.html ... (failure)") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2384 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/2016 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/41308 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitWebContext.cpp:
```diff
@@ -638,35 +639,36 @@ static void testWebContextSecurityFileXHR(WebViewTest* test, gconstpointer)
     test->waitUntilLoadFinished();
 
     GUniquePtr<char> jsonURL(g_strdup_printf("file://%s/simple.json", Test::getResourcesDir().data()));
-    GUniquePtr<char> xhr(g_strdup_printf("var xhr = new XMLHttpRequest; xhr.open(\"GET\", \"%s\"); xhr.send();", jsonURL.get()));
+    GUniquePtr<char> xhr(g_strdup_printf("var xhr = new XMLHttpRequest; xhr.open(\"GET\", \"%s\"); xhr.onreadystatechange = ()=> { if (xhr.readyState == 4) { setTimeout(() => { window.webkit.messageHandlers.xhr.postMessage('DONE'); }, 0)} }; xhr.onerror = () => { window.webkit.messageHandlers.xhr.postMessage('ERROR'); }; xhr.send();", jsonURL.get()));
```
Converting this to a multi-line string would aid readability.

mcatanzaro reviewed (approved):
Normally I would ask why are you removing the implementation, in case somebody is using this that we just don't know about, but I see it's because it's using InjectedBundle. OK then: risk vs. reward seems worth it.

carlosgcampos committed:
[GTK][WPE] Deprecate WebKitConsoleMessage API
https://bugs.webkit.org/show_bug.cgi?id=247791

Reviewed by Michael Catanzaro.

The API seems to be unused so we can also remove the implementation.

* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitConsoleMessage.cpp:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitConsoleMessage.h.in:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebPage.cpp:
(webkit_web_page_class_init):
(webkitWebPageCreate):
(webkitWebPageDidSendConsoleMessage): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestConsoleMessage.cpp: Removed.
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitWebContext.cpp:
(xhrMessageReceivedCallback):
(testWebContextSecurityFileXHR):
(consoleMessageReceivedCallback): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitGLib/WebExtensionTest.cpp:
(pageCreatedCallback):
(consoleMessageSentCallback): Deleted.
* Tools/TestWebKitAPI/glib/CMakeLists.txt:

Canonical link: https://commits.webkit.org/257010@main
